### DISCUSSION TITLE
'server actions' could be destructive in production environment

### DISCUSSION THREAD

**Discussion by FanetheDivine** [vercel/next.js#60024] [Help]
### Summary

I build and start my project in production environment and then I click this button  
```ts
<button onClick={serverAction}>submit</button>
```
and then there is an error page instead of my own  
<img width="1262" alt="1" src="https://github.com/vercel/next.js/assets/141594291/a4760e39-4cd9-4cac-a33f-eb1bbec9431c">
anyway,an error should never perform such a destructive action
could I avoid this problem in production environment?

**Comment by matthewraaff** [comment]
Hi, could we see the output of your browser console?

**Reply by FanetheDivine** [reply]
just one message
<img width="871" alt="屏幕截图 2023-12-29 224710" src="https://github.com/vercel/next.js/assets/141594291/4ca0806d-761d-48c5-9e1c-1d144645812a">

**Reply by matthewraaff** [reply]
@FanetheDivine It's hard to understand what's happening in your application without seeing the function of serverAction, but there are a few things you can try. This [reddit thread](https://www.reddit.com/r/nextjs/comments/17rnx8a) has an answer and good explanation. You can change the button to something such as `<button onClick={e => serverAction({})}>submit</button>`. If this doesn't work, please respond with the code behind serverAction. Thanks.

**Reply by FanetheDivine** [reply]
`serverAction` doesn't matter because no network request is sended and the function is not executed. 
Of course there is no error if correct parameters provided, but I prefer that this error should not take the page out of developers' control.
By the way, I tried to catch the error but completely failed.
